superset/sqllab/sqllab_execution_context.py

```diff
@@ -76,7 +76,7 @@ def _init_from_query_params(self, query_params: Dict[str, Any]) -> None:
         self.sql = cast(str, query_params.get("sql"))
         self.template_params = self._get_template_params(query_params)
         self.async_flag = cast(bool, query_params.get("runAsync"))
-        self.limit = self._get_limit_param(query_params)
+        self.limit = self._get_limit_param(self.sql, query_params)
```

Comment:
What's the root cause here, why are we not passing a limit in `query_params`?

Reply:
Query params does have a limit column, but currently that is `queryLimit,` which is what limit that is in the query dropdown.

Reply:
The reason query_params only has access to queryLimit is because it is being instantiated here: https://github.com/preset-io/superset/blob/master/superset/views/core.py#L2393

with the `request.json`, that is why the only limit is queryLimit, because at that point we haven't parsed the SQL yet.

Reply:
Ah, makes sense! Thanks!